src/tests/functional/test_competitions.py

```diff
@@ -44,3 +50,91 @@ def test_upload_v15_competition(self):
 
     def test_upload_v2_competition(self):
         self._upload_competition('competition.zip')
+
+    def test_manual_competition_creation(self):
+
+        # Dataset Creation
+        self.find('i[selenium="tasks"]').click()
+        self.find('div[data-tab="datasets"]').click()
+        self.find('i[selenium="add-dataset"]').click()
+        self.find('input-text[selenium="scoring-name"] input').send_keys('sCoRiNg NaMe')
+        self.find('input-text[selenium="scoring-desc"] input').send_keys('sCoRiNg DeScRiPtItIoN')
+        self.execute_script('$("select[selenium=\'type\']").dropdown("set selected", "scoring_program")')
+        self.find('input-file[selenium="file"] input').send_keys(os.path.join(self.test_files_dir, 'scoring_program.zip'))
+        self.find('i[selenium="upload"]').click()
+
+        sleep(2)
+        # Task Creation
+        self.find('div[data-tab="tasks"]').click()
+        self.find('div[selenium="create-task"]').click()
+        self.find('input[selenium="name2"]').send_keys('nAmE')
+        self.find('textarea[selenium="task-desc"]').send_keys('textbox')
+        self.find('div[data-tab="data"]').click()
+        self.find('input[id="scoring_program"]').send_keys('sco')
+        sleep(.5)
+        self.execute_script('$("div[selenium=\'scoring-program\'] a")[0].click()')
+        self.find('div[selenium="save-task"]').click()
+
+        # Details Tab
+        competition_title = "selenium_test_comp"
+        self.get(reverse('competitions:create'))
+        self.find('input[ref="title"]').send_keys(competition_title)
+        self.find('input[ref="logo"]').send_keys(os.path.join(self.test_files_dir, '3.png'))
+        self.find('input[ref="docker_image"]').send_keys('docker_image')
+
+        # Participation Tab
+        self.find('a[data-tab="participation"]').click()
+        self.execute_script('$("textarea[ref=\'terms\']")[0].EASY_MDE.value("pArTiCiPaTe")')
+        sleep(2)
+        self.find('input[selenium="auto-approve"]').click()
+
+        # Pages Tab
```

Comment:
Nice comments and structure

Reply:
Sweet thanks. I thought it made it easier to parse through.